platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/resolve/internal/ArtifactRepositoriesPluginResolver.java

```diff
@@ -175,13 +180,17 @@ private ModuleDependency getMarkerDependency(PluginRequestInternal pluginRequest
         ModuleVersionSelector selector = pluginRequest.getModule();
         if (selector == null) {
             String id = pluginRequest.getId().getId();
-            return new DefaultExternalModuleDependency(id, id + PLUGIN_MARKER_SUFFIX, pluginRequest.getVersion());
+            return getDependencyFactory().create(id, id + PLUGIN_MARKER_SUFFIX, pluginRequest.getVersion());
         } else {
-            return new DefaultExternalModuleDependency(selector.getGroup(), selector.getName(), selector.getVersion());
+            return getDependencyFactory().create(selector.getGroup(), selector.getName(), selector.getVersion());
         }
     }
 
     private String getNotation(Dependency dependency) {
         return Joiner.on(':').join(dependency.getGroup(), dependency.getName(), dependency.getVersion());
     }
+
+    private DependencyFactory getDependencyFactory() {
```

Comment:
🤔 I think it would be better to story the `DependencyFactory` as a field in the constructor, rather than having to call this `private` getter everytime.  That would also remove the need for the `resolutionServices` field.

Reply:
ResolutionServices is already used elsewhere in this class. Do you recommend pulling out the other used field as well?

Reply:
I missed it being used in the folded unchanged section.  Probably not worth removing removing the one field to add two others then.  Still, I think you can avoid the getter.